cluster/engine.go

```diff
@@ -217,8 +217,13 @@ func (e *Engine) StartMonitorEvents() {
 		if err := <-ec; err != nil {
 			log.WithFields(log.Fields{"name": e.Name, "id": e.ID}).WithError(err).Error("error monitoring events, will restart")
 			if !strings.Contains(err.Error(), "EOF") {
-				// failing node reconnect should use back-off strategy
-				<-e.refreshDelayer.Wait(e.getFailureCount())
+				// failing node reconnect should use back-off strategy to avoid frequent reconnect
+				retryInterval := e.getFailureCount() + 1
```

Comment:
I believe that, by default, `retryInterval` is always going to be at least 30, right?

https://github.com/docker/swarm/blob/master/cli/flags.go#L71

So with the default settings, this will just always be set to 10.